### ISSUE TITLE
getChannelNumber() ist not updating when communication channel is switched

### ISSUE THREAD

**Issue by saege8** [dji-sdk/Mobile-SDK-Android#445]
Hi, I am trying to access the currently used channel number via the mobile SDK (Android). I am following the example code as seen below and it has been tested with a Phantom 3 so far. The code is working and it provides successful feedback with the correct channel number when the system is initialized. However, if the system is switching the channel, the SDK is not reporting the new channel but still report that the initial one is still used.

If the app is restarted and a new connection is established, the SDK reports the correct value until the channel is changed again (keeps reporting the original one). 

```
XXX.getProductInstance()
                    .getAirLink()
                    .getLightbridgeLink()
                    .getChannelNumber(new CommonCallbacks.CompletionCallbackWith<Integer>() {

                        @Override
                        public void onSuccess(Integer integer) {
                            //  ----->   print out channel no
                        }

                        @Override
                        public void onFailure(DJIError djiError) {

                        }
                    });
        }
```

So far it has been tested with a Phantom 3.

Does anyone have an explanation for this behaviour? Is there some kind of misunderstanding on our side how to use the getChannelNumber Method?

Thanks!

**Comment by dji-dev** [comment]
Public comment from Lisa Fedane in <a href='https://djisdksupport.zendesk.com/agent/tickets/28570'>Zendesk ticket #28570</a>:<br><br>In your code sample this is a single use, get function and because of that, the callback is no longer active after the initial reading. So if you set the channel number to something else later on, then you will need to get it again as well.

Lisa Fedane
DJI Developer Support

**Comment by saege8** [comment]
Yes, you are right. The thing is that this code is called on a regular basis. First thing was to request the channel number by creating a setDownlinkSignalQualityCallback and using its onUpdate method (calling the above code on each update). This did not work, as it seems that the update is happening only once (maybe this is an related issue). I would have expected that this method always fire if the system detects a change in the link quality?

Second approach was to call the code each time I get a position update. So the getChannelNumber is called really often. But the channel reported by the sdk is not changing even if the system jumps to a new one.
We did some experiments with other get methods from the sdk like getDataRate(). This worked perfectly well and we received changing feedback (for example if the connection gets worse) but neither the onUpdate Method of setDownlinkSignalQualityCallback() did fire nor did the return value of the getChannelNumber method did change.

Last idea was to put the code inside an own thread to call it separately but as the getDataRate() Method was working without issues, I fear that this would not lead to better results.

So, is there any other guess?

Thanks!
Florian
